### ISSUE TITLE
Migrate RoboGofer and RoboWitness to BCGov Account Holders

### ISSUE THREAD

**Issue by j-pye** [BCDevOps/developer-experience#2140]
RoboWitness and RoboGofer are machine accounts (read: standard github accounts for machine use) meant to assist in the automation and use of ArgoCD for the platform-services team and tenants using GitOps. (The removal of these accounts from bcgov-c, bcdevops, or bcgov-platform-services will result in every gitops system breaking such as CCM, Project Registry, Tenant GitOps, etc)

These accounts are currently linked to Justin Pye's email and 2FA. Since I, Justin Pye, am no longer a full time member of the team these accounts need to be migrated to other platform services controlled email addresses. 

**Migration Options:**
1. Create 1 PS Team gmail account, use `+` email aliases to link both github accounts to this email, work with 1 or more team members to migrate 2FA / Authy (mobile app)
2. Create 2 BCGov email addresses, 1 for each account, Migrate 2FA for each account

**Note:** Shared 2FA with 1password would be much easier if it's available. 

**Remove RoboWitness from the bcgov-c inactive list permanently as it will never show activity. -- Reading does not count towards GH activity.**

**Comment by j-pye** [comment]
>Remove RoboWitness from the bcgov-c inactive list permanently as it will never show activity. -- Reading does not count towards GH activity.

Context: https://github.com/bcgov-c/platform-service-notification/issues/14

**Comment by mitovskaol** [comment]
@j-pye Does it have to be a "gmail" account? We have a team's mailbox PlatformServicesTeam@gov.bc.ca that is linked to IDIR, if we can enabled 2FA for its IDIR, maybe we can use this email instead?

**Comment by j-pye** [comment]
@mitovskaol There are 2 github accounts and each one needs a unique email address. Gmail and pretty much every other email provider other than outlook@bcgov allows you to do something like myemail+robogofer@gmail.com and myemail+robowitness@gmail.com and have the mail go to myemail@gmail.com. Currently, RoboGofer and RoboWitness are using a `+` alias to my email. 

TL;DR if you want to use @gov.bc.ca emails then you will need two. 

For the 2FA piece, there will be 1x  2FA config per account whether it's gmail or a bcgov address. The reason I mentioned 1password is because you can configure 2FA in 1password for an account and have that be apart of the shared credentials. Without a shared password manager like 1password, every person who needs to access the github accounts will need to setup and configure a 2FA authenticator per github account or contact the account owner and have them share the 2FA code in real time.

**Comment by j-pye** [comment]
I should add that I have tried the `+` modifier to enable an alias with a bcgov email address and it does not work. There may be something internal that the team that manages the bcgov email service can do to setup another unique address that goes to the same inbox. It's not something enabled by default though and when I asked 8+ months ago I was told it was not an option.

**Comment by lukegonis** [comment]
@j-pye Hey Justin, can you pleas provide an update on this issue?

**Comment by j-pye** [comment]
@lukegonis @mitovskaol The accounts have been migrated to the new emails 
github+robogofer-at-bcdevexchange.org
github+witness-at-bcdevexchange.org

Which is technically 1 email. github-at-bcdevexchange.org

We still need to transfer the github accounts to the team though and since they need to have 2FA enabled this is quite the challenge. If 1password isn't ready, I could create an https://authy.com/ account for the team to share and store the credentials in Vault. Then each member of the team with access to the shared Authy account can login to auth to get the 2FA token for the github accounts. I don't know if another way to share 2FA access.

How does that sound? It's either that or we limit access to a single person and we can meet to do the transfer?

**Comment by StevenBarre** [comment]
> I don't know if another way to share 2FA access.

Our team just installed the Google Auth app and all imported the QR code. So we can all log into the one advsol test github account.

**Comment by mitovskaol** [comment]
@NickCorcoran What is your advice re: where to keep these passwords?

**Comment by NickCorcoran** [comment]
As @j-pye and @StevenBarre alluded to, it's not just about where to keep the passwords, but how to access them.  I'm happy with them being stored in Vault or as GitHub secrets for the team to access.  We just have to ensure those that need to know have the ability to get them (@ShellyXueHan @caggles).  I think it's more important that we transfer them into a useable place, and then figure out the longer term team password sharing strategy later (and hopefully that turns out to be 1Password in the somewhere-near future).

**Comment by mitovskaol** [comment]
@j-pye As few of us have set up 2FA for these robo accounts on our smartphones, can this ticket be closed?

**Comment by j-pye** [comment]
@mitovskaol This can close when this is complete.
https://app.zenhub.com/workspaces/platform-experience-5bb7c5ab4b5806bc2beb9d15/issues/bcdevops/developer-experience/2230
